source/plugins/go/src/oms.go

```diff
@@ -1049,6 +1053,25 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 	for _, record := range tailPluginRecords {
 		containerID, k8sNamespace, k8sPodName, containerName := GetContainerIDK8sNamespacePodNameFromFileName(ToString(record["filepath"]))
 		logEntrySource := ToString(record["stream"])
+		logEntry := ToString(record["log"])
+		logEntryTimeStamp := ToString(record["time"])
+
+		if (stripMultilineHeaders) {
+			logEntry = multilineContainerdRegex.ReplaceAllString(logEntry, "\n")
+
+			// If there's a single line log when multiline is enabled, it won't be parsed by the parser
+			// and the whole line with the containerd header will be the log entry.
+			// Parse the timestamp, source, and log entry, so the log isn't dropped in the LA pipeline.
+			if len(logEntryTimeStamp) == 0 {
+				var containerdHeaderRegex = regexp.MustCompile(`^([^ ]+) (stdout|stderr) ([^ ]*) ( *(.*))$`)
+				groupMatches := containerdHeaderRegex.FindStringSubmatch(logEntry)
```

Comment:
FindStringSubmatch can return "nil"- https://golang.org/src/regexp/regexp.go. From the reading the golang docs, len of ni slice is 0 - https://tour.golang.org/moretypes/12. I am wondering, we should add "nil" check to make sure we dont crash if the implementation of this function changes with go version.  This is not critical but i am thinking as safetynet.